lib/IRGen/GenMeta.cpp

```diff
@@ -2820,6 +2822,19 @@ void irgen::emitStructMetadata(IRGenModule &IGM, StructDecl *structDecl) {
 
   IGM.defineTypeMetadata(declaredType, isIndirect, isPattern,
                          canBeConstant, init.finishAndCreateFuture());
+
+  IGM.IRGen.noteUseOfAnyParentTypeMetadata(structDecl);
+}
+
+void IRGenerator::noteUseOfAnyParentTypeMetadata(NominalTypeDecl *type) {
+  // If this is a nested type we also potentially might need the outer types.
+  auto *declCtxt = type->getDeclContext();
+  assert(declCtxt);
+  auto *parentNominalDecl = dyn_cast_or_null<NominalTypeDecl>(declCtxt);
```

Comment:
On the other hand, this will fail if the type is defined inside an extension. Instead, use `declCtxt->getAsNominalTypeOrNominalTypeExtensionContext()`.

Reply:
Will do.